sqlmesh/core/plan/definition.py

```diff
@@ -441,7 +442,7 @@ def has_unmodified_unpromoted(self) -> bool:
 
     def _add_restatements(self) -> None:
         def is_restateable_snapshot(snapshot: Snapshot) -> bool:
-            if not self.is_dev and snapshot.model.disable_restatement:
+            if not self.is_dev and snapshot.is_model and snapshot.model.disable_restatement:
```

Comment:
I feel like `disable_restatement` should be a part of `Snapshot` and look like this:
```
return self.is_model and self.model.disable_restatement
```